### PR TITLE
fix: update type definition file (fix #383)

### LINKED ISSUE #383: export interfaces in typescript declarations
## Version
v1.12.4

## Development Environment
Windows 11, Chrome

## Current Behavior

```
declare module 'tui-calendar' {
    export default tuiCalendar.Calendar;
}
```

Is increadibly restrictive. 

## Expected Behavior
It would be good to export interfaces too, so that we could write types that use them, like for instance for https://github.com/nhn/toast-ui.react-calendar

### PR DESCRIPTION
**Pull request by jungeun-cho**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description
* update type definition file

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Comment by jungeun-cho** [comment]
ok to test

File: index.d.ts [hunk 1]
```diff
@@ -1,294 +1,276 @@
...
 
-    class TZDate {
-        public getTime(): number;
-        public toDate(): Date;
-        public toUTCString(): Date;
-    }
+export class TZDate {
+    public getTime(): number;
+    public toDate(): Date;
+    public toUTCString(): Date;
```
Comment (dongsik-yoo) [inline index.d.ts]: This return type must be `string`. And check and fix the JSDoc at timezone.js please.

File: index.d.ts [hunk 2]
```diff
@@ -1,294 +1,276 @@
...
-        start: TZDate;
-        calendar?: ICalendarInfo;
-        triggerEventName?: string;
-    }
+export interface IEventObject {
+    schedule: ISchedule;
+    end: TZDate;
+    start: TZDate;
+    calendar?: ICalendarInfo;
+    triggerEventName?: string;
```
Comment (dongsik-yoo) [inline index.d.ts]: triggerEventName must have union string type. Like `'click' | 'dblclick'`.

File: index.d.ts [hunk 3]
```diff
@@ -1,294 +1,276 @@
...
+        }
+    };
+    scheduleFilter?: (schedule: ISchedule) => boolean;
+}
 
-    interface IRaw {
-        [propName: string]: string | number | boolean | object | null;
-    }
+export interface IRaw {
+    [propName: string]: string | number | boolean | object | null;
```
Comment (dongsik-yoo) [inline index.d.ts]: Let's use [Record<K, T>](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt) in v2.

File: index.d.ts [hunk 4]
```diff
@@ -1,294 +1,276 @@
...
+    isFocused?: boolean;
+    isVisible?: boolean;
+    isReadOnly?: boolean;
+    isPrivate?: boolean;
+    color?: string;
+    bgColor?: string;
+    dragBgColor?: string;
+    borderColor?: string;
+    customStyle?: string;
+    raw?: any;
```
Comment (dongsik-yoo) [inline index.d.ts]: raw type is `IRaw`.

Comment (dongsik-yoo) [reply index.d.ts]: If not, I think IRaw is not necessary.

File: index.d.ts (hunk 3 above)
Comment (jungeun-cho) [reply index.d.ts]: 👍

File: index.d.ts (hunk 4 above)
Comment (jungeun-cho) [reply index.d.ts]: ok. remove `IRaw` types

File: index.d.ts (hunk 1 above)
Comment (jungeun-cho) [reply index.d.ts]: right. The return type of toUTCString is `string`.
doc fixed. too

File: index.d.ts (hunk 2 above)
Comment (jungeun-cho) [reply index.d.ts]: ok.

**Pull request merged**